diff --git a/include/test.h b/include/test.h
index f6d5270..cccd6dc 100644
--- a/include/test.h
+++ b/include/test.h
@@ -53,6 +53,7 @@
 
 /* virt types for tst_is_virt() */
 #define VIRT_XEN	1	/* xen dom0/domU */
+#define VIRT_KVM	2	/* only default virtual CPU */
 
 /*
  * Ensure that NUMSIGS is defined.
diff --git a/lib/tst_virt.c b/lib/tst_virt.c
index d1f1e74..87f73dc 100644
--- a/lib/tst_virt.c
+++ b/lib/tst_virt.c
@@ -24,6 +24,28 @@
 
 #include <unistd.h>
 #include "test.h"
+#include "safe_macros.h"
+
+static int is_kvm(void)
+{
+	FILE *cpuinfo;
+	char line[64];
+	int found;
+
+	/* this doesn't work with custom -cpu values, since there's
+	 * no easy, reasonable or reliable way to work around those */
+	cpuinfo = SAFE_FOPEN(NULL, "/proc/cpuinfo", "r");
+	found = 0;
+	while (fgets(line, sizeof(line), cpuinfo) != NULL) {
+		if (strstr(line, "QEMU Virtual CPU")) {
+			found = 1;
+			break;
+		}
+	}
+
+	SAFE_FCLOSE(NULL, cpuinfo);
+	return found;
+}
 
 static int is_xen(void)
 {
@@ -47,6 +69,8 @@ int tst_is_virt(int virt_type)
 	switch (virt_type) {
 	case VIRT_XEN:
 		return is_xen();
+	case VIRT_KVM:
+		return is_kvm();
 	}
 	tst_brkm(TBROK, NULL, "invalid virt_type flag: %d", virt_type);
 }
diff --git a/testcases/kernel/syscalls/getrusage/getrusage04.c b/testcases/kernel/syscalls/getrusage/getrusage04.c
index e540a69..c2f34f4 100644
--- a/testcases/kernel/syscalls/getrusage/getrusage04.c
+++ b/testcases/kernel/syscalls/getrusage/getrusage04.c
@@ -200,8 +200,9 @@ static void setup(void)
 {
 	tst_sig(NOFORK, DEF_HANDLER, cleanup);
 
-	if (tst_is_virt(VIRT_XEN))
-		tst_brkm(TCONF, NULL, "This testcase is not supported on Xen.");
+	if (tst_is_virt(VIRT_XEN) || tst_is_virt(VIRT_KVM))
+		tst_brkm(TCONF, NULL, "This testcase is not supported on this"
+		        " virtual machine.");
 
 	BIAS_MAX = guess_timer_resolution();
 
