diff --git a/testcases/commands/su/su01 b/testcases/commands/su/su01
index dd1517f..3a50376 100755
--- a/testcases/commands/su/su01
+++ b/testcases/commands/su/su01
@@ -171,6 +171,14 @@ do_cleanup() {
	rm -f $TEST_ENV_FILE > /dev/null 2>&1
 }

+
+#-----------------------------------------------------------------------
+# disable screen
+profile_backup=$(mktemp "/etc/profile.XXXXXX") || exit 1
+cp -a /etc/profile "$profile_backup" || exit 1
+sed -i 's/-w $(tty)/-n ""/' /etc/profile
+#-----------------------------------------------------------------------
+
 #-----------------------------------------------------------------------
 # FUNCTION:  MAIN
 #-----------------------------------------------------------------------
@@ -178,4 +186,5 @@ do_setup
 /bin/su $TEST_USER1 -c ${TCbin}/su01_s1
 EXIT_CODE=$?
 do_cleanup
+mv -f "$profile_backup" /etc/profile  # re-enable screen
 exit $EXIT_CODE
